x/marker/keeper/send_restrictions.go

```diff
@@ -146,3 +146,44 @@ func ContainsWildCard(attr string) bool {
 	segs := strings.Split(attr, ".")
 	return len(segs) > 1 && segs[0] == "*"
 }
+
+// AddToRequiredAttributes add new attributes to current list, errors if attribute already exists
+func AddToRequiredAttributes(currentAttrs []string, addAttrs []string) ([]string, error) {
```

Comment:
These two functions should probably be in the `types` package, maybe even as methods on the marker type.